pkg/storage/pebble_mvcc_scanner.go

```diff
@@ -479,6 +492,11 @@ func (p *pebbleMVCCScanner) getAndAdvance() bool {
 		// Note that this will trigger an error higher up the stack. We
 		// continue scanning so that we can return all of the intents
 		// in the scan range.
+		if p.err = p.memMonitor.Grow(
+			context.TODO(), int64(len(p.curRawKey)+len(p.curValue))); p.err != nil {
+			log.Infof(context.TODO(), "intent memory monitor error: %s", p.err.Error())
```

Comment:
ditto

Reply:
Whenever we return an error from the account, we should wrap it with information about the mvccScanner: the number of bytes in the `pebbleResults` and the `p.intents`. I hope the error from the monitor already includes the current allocations and how much was requested. We can then check that these are in line with expectations when we see such an error (and spot bugs if there are any).

I also agree with @jordanlewis that this should be a structured error, so that SQL can annotate it programmatically if desired.